src/rttable.c

```diff
@@ -216,6 +220,47 @@ void clearAllRoutes(void) {
     my_log(LOG_NOTICE, 0, "All routes removed. Routing table is empty.");
 }
 
+/**
+*	Function to find first downstream host slot
+*/
+int countDownstreamHosts(struct RouteTable *croute) {
+    int result = 0;
+    
+    int i;
+    for (i = 0; i < MAX_DOWNSTREAM_HOSTS; i++) {
+        if(croute->dwnstrHosts[i] != 0) {
+            result++;
+        }
+    }
+    return result;
+}
+
+/**
+*	Function to find first downstream host slot
+*/
+int findFreeDownstreamSlot(struct RouteTable *croute) {
+    int i;
+    for (i = 0; i < MAX_DOWNSTREAM_HOSTS; i++) {
```

Comment:
igmpproxy depends on C99 compiler, so you can use `for (size_t i = 0; ...` construct.